app-web/src/store/reducers/siphon.js

```diff
@@ -125,45 +139,27 @@ const applySearchResultsToPrimaryNodes = (state, results) => {
       nodesMap.set(n.parent.id, currentNodes.concat([{ ...n }]));
     }
   });
-  // build filtered nodes back into respective collections
-  newState.collections = newCollections(state._collections).map(c => ({
-    ...c,
-    nodes: nodesMap.get(c.id) || [],
-  }));
-  return applySecondaryFilters(newState);
-};
-
-/**
- * Check the number of resources that match a filter
- * and applies count related props to the filter Map
- * @param {Object} filter
- * @param {Array} primaryFilteredNodes
- */
-export const applyPropsToFilterByResourceCount = (filter, primaryFilteredNodes) => {
-  // get a flattened set of the nodes from all collections to loop over
-  const nodes = getAllNodesFromCollections(primaryFilteredNodes);
-  const { filterBy, value } = filter;
-  let newFilter = { ...filter, availableResources: 0 };
 
-  nodes.forEach(n => {
-    newFilter.availableResources += dotPropMatchesValue(n, filterBy, value);
+  // a store of all nodes that passed the filter this is passed into the apply props to filter fn
+  // as an optimization. Soon to be added is a caching mechanism where search terms and newstate collections
+  // are cached in a map of 'search term': [collections]
+  let availableNodes = [];
+  // build filtered nodes back into respective collections
+  newState.collections = newCollections(state._collections).map(c => {
+    const nodes = nodesMap.get(c.id) || [];
+    // push in the nodes that are being applied to collection into available nodes
+    availableNodes = availableNodes.concat(nodes);
```

Comment:
why need `availableNodes` since you already putting nodes in the collection?

Reply:
available nodes is the entire list of nodes that is passed into applyPropsToFilterByResourceCount